viewer/packages/tools/src/AxisGizmo/AxisGizmoTool.ts

```diff
@@ -0,0 +1,443 @@
+/*
+ * Copyright 2024 Cognite AS
+ */
+
+import { Matrix4, PerspectiveCamera, Vector3 } from 'three';
+import { AxisGizmoOptions } from './AxisGizmoOptions';
+import { CDF_TO_VIEWER_TRANSFORMATION } from '@reveal/utilities';
+import { Cognite3DViewer } from '@reveal/api';
+import { OneGizmoAxis } from './OneGizmoAxis';
+import { CameraManager, IFlexibleCameraManager } from '@reveal/camera-manager';
+import { moveCameraTo } from '../utilities/moveCameraTo';
+import { Corner } from '../utilities/Corner';
+import { Cognite3DViewerToolBase } from '../Cognite3DViewerToolBase';
+
+/**
+ * Class for axis gizmo like the one in Blender
+ * @beta
+ */
+export class AxisGizmoTool extends Cognite3DViewerToolBase {
+  //================================================
+  // INSTANCE FIELDS
+  //================================================
+  private readonly _options: AxisGizmoOptions;
+  private readonly _axises: OneGizmoAxis[];
+  private readonly _center: Vector3;
+  private _mousePosition: Vector3 | undefined = undefined;
+  private _selectedAxis: OneGizmoAxis | undefined = undefined;
+  private _isMouseOver = false; // Keep track of this for highlighing the gizmo
+  private _inDragging = false;
+
+  private _viewer: Cognite3DViewer | null = null;
+  private _element: HTMLElement | null = null;
+  private _canvas: HTMLCanvasElement | null = null;
+  private _context: CanvasRenderingContext2D | null = null;
+
+  // binding of the event functions:
+  private readonly _onPointerDown = this.onPointerDown.bind(this);
+  private readonly _onPointerUp = this.onPointerUp.bind(this);
+  private readonly _onPointerMove = this.onPointerMove.bind(this);
+  private readonly _onPointerOut = this.onPointerOut.bind(this);
+  private readonly _onMouseClick = this.onMouseClick.bind(this);
+  private readonly _onMouseDoubleClick = this.onMouseDoubleClick.bind(this);
+
+  //================================================
+  // CONSTRUCTORS
+  //================================================
+
+  constructor() {
```

Comment:
I'm not sure I like that the tool is constructed first, and then connected to the 3D-viewer. It should not be able to exist without the 3D-viewer, right? It would make it easier not having to keep track of an uninitialized state

Reply:
That is the meaning. It it a pattern than used everywhere else than Reveal. For instance in Tree.js. There you create an object and you can add it later .